python/pyspark/sql/readwriter.py

```diff
@@ -563,6 +563,60 @@ def partitionBy(self, *cols):
         self._jwrite = self._jwrite.partitionBy(_to_seq(self._spark._sc, cols))
         return self
 
+    @since(2.3)
+    def bucketBy(self, numBuckets, *cols):
+        """Buckets the output by the given columns on the file system.
```

Comment:
I'd copy the full description from DataFrameWriter here since comparing it to Hive could help people new to Spark understand what bucketBy does.